web_external/js/views/map/MapAdapters.js

```diff
@@ -0,0 +1,212 @@
+minerva.core = minerva.core || {};
+
+(function () {
+    function AdapterRegistry() {
+        this.registry = {};
+        this.register = function (type, definition) {
+            this.registry[type] = definition;
+        };
+        this._createRepresentation = function (container, dataset, layerType, mapping) {
+            if (layerType === null || !_.has(this.registry, layerType)) {
+                console.error('This dataset cannot be adapted to a map layer of type [' + layerType + '].');
+                this.trigger('m:map_adapter_error', dataset, layerType);
+                return;
+            } else {
+                var adapter = this.registry[layerType];
+                var layerRepr = _.extend(new adapter(), Backbone.Events);
```

Comment:
@jbeezley Here's where I brought in Backbone.Events, suggestions welcome for a simpler architecture.